### ISSUE TITLE
Common table expression in subquery

### ISSUE THREAD

**Issue by erw6f** [oguimbal/pgsql-ast-parser#18]
Using a CTE within a subquery is not accounted for in the parser:
```sql
SELECT
  regexp_replace(
    (
      SELECT
        (
          WITH mycte AS (
            SELECT
              myctecol
            FROM
              myschema.mytable
            WHERE
              mycolumn like 'LIKETHIS%'
              AND mything = $3
            ORDER BY
              ordering
          )
          SELECT
            string_agg(myctecol, '; ')
          FROM
            mycte
        )
    ),
    '/regex/',
    'replaced',
    'g'
  )
```
Error:
```
  SELECT regexp_replace((SELECT (WITH
                                 ^
Unexpected kw_with token: "with". Instead, I was expecting to see one of the following:
```

**Comment by oguimbal** [comment]
Hi !

Yup, this syntax is not supported.
Thanks for there report, I'll implement it when i find some time, hopefully next week :)

**Comment by oguimbal** [comment]
Fixed with `pgsql-ast-parser@5.1.0`
